cmd/npmExecuteLint.go

```diff
@@ -0,0 +1,169 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/SAP/jenkins-library/pkg/command"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/npm"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+type lintUtils interface {
+	Glob(pattern string) (matches []string, err error)
+
+	getExecRunner() execRunner
+	getGeneralPurposeConfig(configURL string) error
+}
+
+type lintUtilsBundle struct {
+	*piperutils.Files
+	execRunner *command.Command
+	client     *piperhttp.Client
+}
+
+func newLintUtilsBundle() *lintUtilsBundle {
+	return &lintUtilsBundle{
+		Files:  &piperutils.Files{},
+		client: &piperhttp.Client{},
+	}
+}
+
+func (u *lintUtilsBundle) getExecRunner() execRunner {
+	if u.execRunner == nil {
+		u.execRunner = &command.Command{}
+		u.execRunner.Stdout(log.Writer())
+		u.execRunner.Stderr(log.Writer())
+	}
+	return u.execRunner
+}
+
+func (u *lintUtilsBundle) getGeneralPurposeConfig(configURL string) error {
+	response, err := u.client.SendRequest(http.MethodGet, configURL, nil, nil, nil)
+	if err != nil {
+		return err
+	}
+
+	defer response.Body.Close()
+
+	content, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return fmt.Errorf("error reading %v: %w", response.Body, err)
+	}
+
+	err = u.FileWrite(filepath.Join(".pipeline", ".eslintrc.json"), content, os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("failed to write .eslintrc.json file to .pipeline/: %w", err)
+	}
+
+	return nil
+}
+
+func npmExecuteLint(config npmExecuteLintOptions, telemetryData *telemetry.CustomData) {
+	utils := newLintUtilsBundle()
+	npmExecutorOptions := npm.ExecutorOptions{RunScripts: []string{"ci-lint"}, RunOptions: []string{"--silent"}, DefaultNpmRegistry: config.DefaultNpmRegistry, SapNpmRegistry: config.SapNpmRegistry, ExecRunner: utils.getExecRunner()}
+	npmExecutor, err := npm.NewExecutor(npmExecutorOptions)
+
+	err = runNpmExecuteLint(npmExecutor, utils, &config)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runNpmExecuteLint(npmExecutor npm.Executor, utils lintUtils, config *npmExecuteLintOptions) error {
+	packageJSONFiles := npmExecutor.FindPackageJSONFiles()
+	packagesWithCiLint, _ := npmExecutor.FindPackageJSONFilesWithScript(packageJSONFiles, "ci-lint")
+
+	if len(packagesWithCiLint) > 0 {
+		err := runCiLint(npmExecutor, config.FailOnError)
+		if err != nil {
+			return err
+		}
+	} else {
+		err := runDefaultLint(npmExecutor, utils, config.FailOnError)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func runCiLint(npmExecutor npm.Executor, failOnError bool) error {
+	err := npmExecutor.ExecuteAllScripts()
```

Comment:
Naming: 
Maybe runScriptsInAllPackages runScriptsInAllModules

Reply:
Exactly. But yeah you are right the current naming is not that good. I would opt for renaming it to runScriptsInAllPackages then.